src/SpecialRequestAccount.php

```diff
@@ -409,13 +434,11 @@ function handleAuthenticationFormSubmission(IDatabase $dbr) {
 			return;
 		}
 
-		$linkRenderer = $this->getLinkRenderer();
-
 		$username = $request->getText('username');
 		$password = $request->getText('password');
 
 		//see if there are any requests with the given password
-		$passwordFactory = MediaWikiServices::getInstance()->getPasswordFactory();
+		$passwordFactory = $this->passwordFactory;
```

Comment:
Notpicking a bit here but I think this line is unnecessary, you can just use `$this->passwordFactory` further on down